diff -ur ffmpeg.orig/ ffmpeg.new/
--- ffmpeg.orig/libavcodec/libxvidff.c	2010-04-20 23:45:34 +0900
+++ ffmpeg/libavcodec/libxvidff-new.c	2010-05-17 04:34:34 +0900
@@ -408,6 +408,11 @@
     xvid_enc_frame.motion = x->me_flags;
     xvid_enc_frame.type = XVID_TYPE_AUTO;
 
+    /* Fix default aspect ratio */
+    if (avctx->sample_aspect_ratio.num == 0 && avctx->sample_aspect_ratio.den == 1) {
+        avctx->sample_aspect_ratio.num = 1;
+    }
+
     /* Pixel aspect ratio setting */
     if (avctx->sample_aspect_ratio.num < 1 || avctx->sample_aspect_ratio.num > 255 ||
         avctx->sample_aspect_ratio.den < 1 || avctx->sample_aspect_ratio.den > 255) {
